sdk/core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java

```diff
@@ -112,6 +131,23 @@ public void info(String message) {
         }
     }
 
+    /**
+     * Logs a message at {@code info} log level.
+     *
+     * <p><strong>Code samples</strong></p>
+     *
+     * <p>Logging a message at information log level.</p>
+     *
+     * {@codesnippet com.azure.core.util.logging.clientlogger.info}
+     *
+     * @param message a supplier for for logging messages.
+     */
+    public void info(Supplier<String> message) {
```

Comment:
We may need to change the names of the new APIs, if someone is was previously passing null into the String based overload they will now run into an ambiguous call compilation error.

Thoughts of `deferredInfo`, or something else along those lines?

Reply:
I was thinking of delayedInfo, but I think deferred is a bit better for this.